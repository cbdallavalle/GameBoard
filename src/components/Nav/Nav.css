nav {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  height: fit-content;
  margin-top: 20px;
  background-color: white;
  padding: 0 20px;
  box-sizing: border-box;
}

nav button {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  border: none;
  padding: 10px;
  background-color: unset;
}

nav button:hover {
  background-color: #d5f0f7;
}

.selected {
  background-color: #bbdde6;
}