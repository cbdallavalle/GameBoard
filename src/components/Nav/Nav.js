import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Nav.css';

export const Nav = () => {
  return (
    <nav>
      <NavLink 
        to='/dashboard/your-games'
        activeClassName="selected"
      >
        <button name='your-games'>
          <i className="fas fa-user"></i> Your Games
        </button>
      </NavLink>
      <NavLink 
        to='/dashboard/friends-games'
        activeClassName="selected"
      >
        <button name='friends-games' >
          <i className="fas fa-users"></i> Your friend's games
        </button>
      </NavLink>
      <NavLink 
        to='/dashboard/search-games'
        activeClassName="selected"
      >
        <button name='search-games'>
          Search for games 
        </button>
      </NavLink>
      <NavLink 
        to='/dashboard/search-users'
        activeClassName="selected"
      >
        <button name="search-friends">
          Search for friends
        </button>
      </NavLink>
    </nav>
  )
}