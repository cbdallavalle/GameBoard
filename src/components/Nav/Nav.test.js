/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';

describe("Nav", () => {
  const mockHandleClick = jest.fn();

  it('should match the snapshot', () => {
    const wrapper = shallow(<Nav handleClick={mockHandleClick} />);
    expect(wrapper).toMatchSnapshot();
  })
})