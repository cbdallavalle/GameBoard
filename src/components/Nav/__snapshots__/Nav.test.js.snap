// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="selected"
          ariaCurrent="true"
          to="/dashboard/your-games"
>
          <button
                    name="your-games"
          >
                    <i
                              className="fas fa-user"
                    />
                    Your Games
          </button>
</NavLink>,
        <NavLink
          activeClassName="selected"
          ariaCurrent="true"
          to="/dashboard/friends-games"
>
          <button
                    name="friends-games"
          >
                    <i
                              className="fas fa-users"
                    />
                    Your friend's games
          </button>
</NavLink>,
        <NavLink
          activeClassName="selected"
          ariaCurrent="true"
          to="/dashboard/search-games"
>
          <button
                    name="search-games"
          >
                    Search for games
          </button>
</NavLink>,
        <NavLink
          activeClassName="selected"
          ariaCurrent="true"
          to="/dashboard/search-users"
>
          <button
                    name="search-friends"
          >
                    Search for friends
          </button>
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "selected",
          "ariaCurrent": "true",
          "children": <button
            name="your-games"
>
            <i
                        className="fas fa-user"
            />
            Your Games
</button>,
          "to": "/dashboard/your-games",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-user"
/>,
              "Your Games",
            ],
            "name": "your-games",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-user",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "Your Games",
          ],
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "selected",
          "ariaCurrent": "true",
          "children": <button
            name="friends-games"
>
            <i
                        className="fas fa-users"
            />
            Your friend's games
</button>,
          "to": "/dashboard/friends-games",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-users"
/>,
              "Your friend's games",
            ],
            "name": "friends-games",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-users",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "Your friend's games",
          ],
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "selected",
          "ariaCurrent": "true",
          "children": <button
            name="search-games"
>
            Search for games
</button>,
          "to": "/dashboard/search-games",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for games",
            "name": "search-games",
          },
          "ref": null,
          "rendered": "Search for games",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "selected",
          "ariaCurrent": "true",
          "children": <button
            name="search-friends"
>
            Search for friends
</button>,
          "to": "/dashboard/search-users",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for friends",
            "name": "search-friends",
          },
          "ref": null,
          "rendered": "Search for friends",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="selected"
            ariaCurrent="true"
            to="/dashboard/your-games"
>
            <button
                        name="your-games"
            >
                        <i
                                    className="fas fa-user"
                        />
                        Your Games
            </button>
</NavLink>,
          <NavLink
            activeClassName="selected"
            ariaCurrent="true"
            to="/dashboard/friends-games"
>
            <button
                        name="friends-games"
            >
                        <i
                                    className="fas fa-users"
                        />
                        Your friend's games
            </button>
</NavLink>,
          <NavLink
            activeClassName="selected"
            ariaCurrent="true"
            to="/dashboard/search-games"
>
            <button
                        name="search-games"
            >
                        Search for games
            </button>
</NavLink>,
          <NavLink
            activeClassName="selected"
            ariaCurrent="true"
            to="/dashboard/search-users"
>
            <button
                        name="search-friends"
            >
                        Search for friends
            </button>
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "selected",
            "ariaCurrent": "true",
            "children": <button
              name="your-games"
>
              <i
                            className="fas fa-user"
              />
              Your Games
</button>,
            "to": "/dashboard/your-games",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-user"
/>,
                "Your Games",
              ],
              "name": "your-games",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-user",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Your Games",
            ],
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "selected",
            "ariaCurrent": "true",
            "children": <button
              name="friends-games"
>
              <i
                            className="fas fa-users"
              />
              Your friend's games
</button>,
            "to": "/dashboard/friends-games",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-users"
/>,
                "Your friend's games",
              ],
              "name": "friends-games",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-users",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Your friend's games",
            ],
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "selected",
            "ariaCurrent": "true",
            "children": <button
              name="search-games"
>
              Search for games
</button>,
            "to": "/dashboard/search-games",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search for games",
              "name": "search-games",
            },
            "ref": null,
            "rendered": "Search for games",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "selected",
            "ariaCurrent": "true",
            "children": <button
              name="search-friends"
>
              Search for friends
</button>,
            "to": "/dashboard/search-users",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search for friends",
              "name": "search-friends",
            },
            "ref": null,
            "rendered": "Search for friends",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
